src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs

```diff
@@ -0,0 +1,618 @@
+// air from the bleed system, and exhausts it to the
+// ambient atmosphere. This is just the implementation
+// of a regular container
+pub struct WingAntiIceConsumer {
+    pipe: PneumaticPipe,
+}
+
+impl PneumaticContainer for WingAntiIceConsumer {
+    fn pressure(&self) -> Pressure {
+        self.pipe.pressure()
+    }
+
+    fn volume(&self) -> Volume {
+        self.pipe.volume()
+    }
+
+    fn temperature(&self) -> ThermodynamicTemperature {
+        self.pipe.temperature()
+    }
+
+    fn mass(&self) -> Mass {
+        self.pipe.mass()
+    }
+
+    fn change_fluid_amount(
+        &mut self,
+        fluid_amount: Mass,
+        fluid_temperature: ThermodynamicTemperature,
+        fluid_pressure: Pressure,
+    ) {
+        self.pipe
+            .change_fluid_amount(fluid_amount, fluid_temperature, fluid_pressure);
+    }
+
+    fn update_temperature(&mut self, temperature_change: TemperatureInterval) {
+        self.pipe.update_temperature(temperature_change);
+    }
+}
+
+impl WingAntiIceConsumer {
+    const CONDUCTION_RATE: f64 = 0.1;
+    pub fn new(volume: Volume) -> Self {
+        Self {
+            pipe: PneumaticPipe::new(
+                volume,
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+        }
+    }
+
+    // Radiate heat to the ambient atmosphere
+    // according to Newton's law of cooling
+    // dT/dt = -(T-T_atmo) / tau
+    pub fn radiate_heat_to_ambient(&mut self, context: &UpdateContext) {
+        let delta_t: TemperatureInterval =
+            TemperatureInterval::new::<temperature_interval::degree_celsius>(
+                self.temperature().get::<degree_celsius>()
+                    - context.ambient_temperature().get::<degree_celsius>(),
+            );
+
+        self.update_temperature(-delta_t * context.delta_as_secs_f64() * Self::CONDUCTION_RATE);
+    }
+}
+
+pub struct WingAntiIceRelay {
+    wing_anti_ice_button_pos: WingAntiIcePushButtonMode, // The position of the button
+    system_test_timer: Duration,                         // Timer to count up to 30 seconds
+    system_test_done: bool, // Timer reached 30 seconds while on the ground
+    signal_on: bool,        // Status of the ON light. If button is pushed and
+    // the test is finished, the ON light should turn off.
+    is_on_ground_id: VariableIdentifier,
+    is_on_ground: bool, //Needed for the 30 seconds test logic
+    powered_by: ElectricalBusType,
+    is_powered: bool,
+}
+impl WingAntiIceRelay {
+    const WAI_TEST_TIME: Duration = Duration::from_secs(30);
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            wing_anti_ice_button_pos: WingAntiIcePushButtonMode::Off,
+            system_test_timer: Duration::from_secs(0),
+            system_test_done: false,
+            signal_on: false,
+            is_on_ground_id: context.get_identifier("SIM ON GROUND".to_owned()),
```

Comment:
If this is the relay that I think it is, then it should indeed use the right hand MLG shock absorber extended signal from LGCIU 2, and the left hand MLG shock absorber extended signal from LGCIU 1, combined basically as an `or` gate

Reply:
I think it uses a discrete ARINC word to determine if the shock absorbers are compressed. Am I correct @lukecologne?

Reply:
No, in this case it is a discrete output of the LGCIU (i.e. one wire, which is either open circuit or connected to ground), so it can close the relay. A429 is only for communication between two computers, since it needs to be received and decoded